### PR TITLE
NO-ISSUE - Add junit reports for download_logs.py

### PR DESCRIPTION
**Pull request by eliorerz**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -421,10 +421,10 @@ delete_all_virsh_resources: destroy_all_nodes delete_minikube kill_all_port_forw
 	skipper run $(SKIPPER_PARAMS) 'discovery-infra/delete_nodes.py -ns $(NAMESPACE) -a'
 
 download_service_logs:
-	./scripts/download_logs.sh download_service_logs
+	JUNIT_REPORT_DIR=$(REPORTS) ${DEBUG_FLAGS} python3 ./scripts/download_logs.sh download_service_logs
 
 download_cluster_logs:
-	./scripts/download_logs.sh download_cluster_logs
+	JUNIT_REPORT_DIR=$(REPORTS) DEBUG_FLAGS="$(DEBUG_FLAGS)" ./scripts/download_logs.sh download_cluster_logs
```
Comment (osherdp) [inline Makefile:427]: why the difference between the lines?
``${DEBUG_FLAGS}`` vs ``DEBUG_FLAGS="$(DEBUG_FLAGS)"``
also ``python3``

File: discovery-infra/download_logs.py [hunk 2]
```diff
@@ -294,6 +300,7 @@ def fetch_url_and_write_to_file(url_key, file_name, debug_info, output_folder):
         log.warning(f"{url_key} is not available")
 
 
+@JunitTestCase()
```
Comment (osherdp) [inline discovery-infra/download_logs.py:303]: why is it a test? :)

File: discovery-infra/download_logs.py [hunk 3]
```diff
@@ -244,6 +249,7 @@ def download_must_gather(kubeconfig: str, dest_dir: str):
         log.warning(f"Failed to run must gather: {ex}")
 
 
+@JunitTestCase()
```
Comment (osherdp) [inline discovery-infra/download_logs.py:252]: should we have ``@JunitTestCase`` per node gathering?

Comment (osherdp) [reply discovery-infra/download_logs.py:252]: unless ``run_concurrently`` makes it a mess to do so

**Review by osherdp — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (eliorerz) [reply Makefile]: By mistake. Fixed

**Comment by YuviGold** [comment]
```
IndexError: list index out of range
```

**Comment by eliorerz** [comment]
New PR - https://github.com/openshift/assisted-test-infra/pull/1029

**Pull request closed without merging**